client/peerset/src/lib.rs

```diff
@@ -272,8 +271,10 @@ impl Peerset {
 		let default_set_config = &config.sets[0];
 		let peer_store = PeerStore::new(default_set_config.bootnodes.clone());
 
-		let (to_notifications, from_controllers) =
-			tracing_unbounded("mpsc_protocol_controllers_to_notifications", 10_000);
+		//let (to_notifications, from_controllers) =
+		//	tracing_unbounded("mpsc_protocol_controllers_to_notifications", 10_000);
+
+		let (to_notifications, from_controllers) = bounded(0);
```

Comment:
What is the plan with rendezvous channel?

Reply:
Basically, it's roughly the same as waiting for ACKs from `Notifications` — we do not do anything else (process incoming actions, allocate slots), until all the outstanding commands for `Notifications` are processes.

I also prioritized polling of connection events from `Notifications` over all other actions, because if we have outstanding commands from `Notifications` and do other things, replies to `Notifications` can be interleaved by other commands and this leads to invalid state transitions.

All this feels hackish, and I still get some invalid event bigrams in the fuzz test, but this invalid events at least can be dealt with (dropped/overrided) at this point. So the fuzz test passes with some adjustments now.

I will look more into invalid bigram issue, may be we can fix it without even more hacks.

Reply:
Alright. My only concern is that if the entire `Peerset`/`ProtocolController` blocks until the response is heard, i.e., if a command is sent for one peer and response must be heard, that peer should be blocked from sending more commands but other peers can still make progress. Not sure if this is already in the implementation.